bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlContentProvider.java

```diff
@@ -33,10 +34,20 @@
 public class TomlContentProvider extends TomlProvider {
 
     private final String configContent;
+    // Finds the `\n` characters that is not inside values to replace with the system line separator
+    private static final Pattern UNESCAPED_NEWLINE_CHAR = Pattern.compile("\\\\n(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
+    // Finds the `\r` and `\t` characters that is not inside values to remove them
```

Comment:
I think these 2 comments go better in the place where we use them, i.e. line 49 and 50.